### PR TITLE
feat: update polymarket notifier logic

### PR DESCRIPTION
**Pull request by md0x**
Proposed changes in this PR include:

- Updating the logic of the Polymarket notifier to initially retrieve markets through on-chain events, and subsequently deduce the corresponding Polymarket market information.
- The fundamental logic for market sentiment notifications remains unchanged.
- This approach enhances resilience against API outages or bugs by ensuring notifications are always linked to an active proposal event.

### CODE REVIEW

File: packages/monitor-v2/src/monitor-polymarket/common.ts [hunk 1]
```diff
@@ -117,135 +66,120 @@ interface PolymarketOrderBook {
...
+
+  const events = await paginatedEventQuery<ProposePriceEvent>(
+    oo,
+    oo.filters.ProposePrice(null, null, null, null, null, null, null, null),
+    searchConfig
+  );
+
+  return events
+    .filter((event) => requesterAddresses.map((r) => r.toLowerCase()).includes(event.args.requester.toLowerCase()))
+    .filter((event) => event.args.expirationTimestamp.gt(BigNumber.from(Math.floor(Date.now() / 1000))))
```
Comment (daywiss) [inline packages/monitor-v2/src/monitor-polymarket/common.ts:122]: is this the correct time to be filtering?

**Review by daywiss — APPROVED** [review]
nice to see so much code removed

File: packages/monitor-v2/src/monitor-polymarket/MonitorLogger.ts [hunk 2]
```diff
@@ -60,68 +58,57 @@ export async function logProposalOrderBook(
...
-    volumeNum: number;
-  }
+  chainId: number,
+  market: OptimisticPriceRequest,
+  error: Error
 ): Promise<void> {
   logger.error({
     at: "PolymarketMonitor",
-    message: "Market proposal event not found for proposed market! 🚨",
+    message: "Failed to verify proposed market, please verify manually! 🚨",
```
[the remark this answers is not in the document]
Comment (md0x) [reply packages/monitor-v2/src/monitor-polymarket/MonitorLogger.ts:101]: It means to check the correctness of the proposal. Do you think my message is unclear, @daywiss?

File: packages/monitor-v2/src/monitor-polymarket/common.ts (hunk 1 above)
Comment (md0x) [reply packages/monitor-v2/src/monitor-polymarket/common.ts:122]: Yes, this retains only the proposals that have an expiration timestamp in the future, meaning they can be disputed and are in their liveness period. The timestamps in the contracts are in seconds; that's why we need to do Date.now()/1000.

File: packages/monitor-v2/src/monitor-polymarket/common.ts [hunk 3]
```diff
@@ -275,200 +208,114 @@ export const getOrderFilledEvents = async (
...
+      .map((event) => getTradeInfoFromOrderFilledEvent(params.provider, event))
   );
+
+  return [outcomeTokenOne, outcomeTokenTwo];
 };
 
-const loadAbi = (filename: string): any => {
-  return require(`./abi/${filename}.json`);
+export const calculatePolymarketQuestionID = (ancillaryData: string): string => {
+  return ethers.utils.keccak256(ancillaryData);
```
Comment (mrice32) [inline packages/monitor-v2/src/monitor-polymarket/common.ts:243]: This is really how the compute question id? I had no idea it was this simple.

Great find!

**Review by mrice32 — APPROVED** [review]
This looks great! I think this methodology change is really great and leans a lot less on non-provider API dependencies.

I also think the general flow of the logic is much more grokable. Great work @md0x!

**Pull request merged**